src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -125,8 +126,43 @@ private StorageListItem GetStorageListItem(IListBlobItem listBlobItem)
                     Trace.WriteLine(String.Format("Saved uncompressed blob {0} to container {1}", blob.Uri.ToString(), _directory.Container.Name));
                 }
             }
+            TakeBlobSnapshot(blob);
         }
 
+        private CloudBlockBlob TakeBlobSnapshot(CloudBlockBlob blob)
+        {
+            if (blob == null)
+            {
+                return blob;
+            }
+            Stopwatch sw = new Stopwatch();
+            Trace.WriteLine($"SnapshotStart:{blob.Uri.ToString()}.");
+            sw.Start();
+            try
+            {
+                //read the current snapshot list ordered decreasing by the SnapshotTime
+                var allSanpshots = blob.Container.ListBlobs(blob.Name, true, blobListingDetails: BlobListingDetails.Snapshots)
+                                   .Where((item) => { return ((CloudBlockBlob)item).IsSnapshot; })
+                                   .OrderBy((item) => { return ((CloudBlockBlob)item).SnapshotTime; });
+
+                if (allSanpshots.Count() >= MaxCountOfBlobSnapshots)
+                {
+                    CloudBlockBlob snapshotToBeDeleted = (CloudBlockBlob)allSanpshots.First();
```

Comment:
minor thing but ideally you'd want to delete all snapshots exceeding `MaxCountofBlobSnapshots` rather than just one. May be not get into this situation if this runs as a single job all the time. But better to put it in here.